intelmq/bots/experts/uri/expert.py

```diff
@@ -0,0 +1,60 @@
+# SPDX-FileCopyrightText: 2022 gutsohnCERT
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# -*- coding: utf-8 -*-
+
+from rfc3986 import uri_reference, validators, exceptions
+from intelmq.lib.bot import ExpertBot
+
+
+class URIExpertBot(ExpertBot):
+    """Parse the URI"""
+    overwrite = False
+
+    def process(self):
+        event = self.receive_message()
+        uri_validator = validators.Validator().require_presence_of('scheme')
+
+        for key in ["source.", "destination."]:
+            key_url = "{}{}".format(key, "url")
+            key_ip = "{}{}".format(key, "ip")
+            key_fqdn = "{}{}".format(key, "fqdn")
+            key_scheme = "{}{}".format(key, "scheme")
+            key_path = "{}{}".format(key, "path")
+            key_userinfo = "{}{}".format(key, "userinfo")
+            key_port = "{}{}".format(key, "port")
+            key_query = "{}{}".format(key, "query")
```

Comment:
```suggestion
            key_url = f"{key}url"
            key_ip = f"{key}ip"
            key_fqdn = f"{key}fqdn"
            key_scheme = f"{key}scheme"
            key_path = f"{key}path"
            key_userinfo = f"{key}userinfo"
            key_port = f"{key}port"
            key_query = f"{key}query"
```

f-strings have been introduced in python3.6 and improve code readability.